### ISSUE TITLE
keybind clash (`super` + `period`) with ibus emoji shortcut and PaperWM `Switch to next/previous window`

### ISSUE THREAD

**Issue by hpaul** [paperwm/PaperWM#690]
**Describe the bug**
After upgrading to Fedora 39 and GNOME 45 I noticed that when I'm moving *from* terminal windows the `Super + .` doesn't behave as expected, somehow is being captured by the shell or by terminal window itself.

This video below should describe the situation better.

**Screenshots**
[Screencast from 2023-11-13 17-06-48.webm](https://github.com/paperwm/PaperWM/assets/594424/e16d44af-408f-43ce-b187-43db83dee44a)

```
Distribution: Fedora 39
GNOME Shell 45
PaperWM branch/tag: release
Enabled extensions:
- paperwm@paperwm.github.com
```

Edit: Forgot to actually add the question.

Do you think that I should debug PaperWM or look somewhere else? As it only happens with only terminal windows and that doesn't help me at all starting with the investigation.

**Comment by Lythenas** [comment]
I can't reproduce this with terminator or gnome terminal. Which terminal are you using?

It is entirely possible that your terminal is intercepting the keybindings. E.g. VMs also do this. Maybe check in the settings for the terminal and disable that keybinding if possible.

**Comment by jtaala** [comment]
Oh, yes, what is that terminal? looks like it's eating keybinds.

**Comment by jtaala** [comment]
> Do you think that I should debug PaperWM or look somewhere else?

Well, definitely test with other gnome terminals (e.g. console or gnome terminal).  If it doesn't happen there for you then most probably that specific terminal application.

**Comment by hpaul** [comment]
@Lythenas 
> I can't reproduce this with terminator or gnome terminal. Which terminal are you using?

I'm using kitty, but it happens on GNOME terminal too.


> It is entirely possible that your terminal is intercepting the keybindings. E.g. VMs also do this. Maybe check in the settings for the terminal and disable that keybinding if possible.

I don't run any VM, but I'll try to disable all keybinding of kitty, see what happens.

----------------
@jtaala 
> Well, definitely test with other gnome terminals (e.g. console or gnome terminal). If it doesn't happen there for you then most probably that specific terminal application.

It happens both on kitty and GNOME Terminal. But I'll try to disable keybinding and see what happens.

**Comment by jtaala** [comment]
hmm - yeah, there must be something else happening here.  I can't reproduce this even with kitty.

https://github.com/paperwm/PaperWM/assets/30424662/76a3cb69-e74f-49d4-a66c-cfb6fbe3e65d

Just checking are you on X11 or wayland?

Also double-checking you have no other extensions enabled?

**Comment by jtaala** [comment]
Interestingly kitty and console does seem to eat keybind `super`+`.`.   I don't use those keybinds (instead using `switch to left/right window` keybinds.

There does seem to be a keybind clash somewhere either in gnome or terminals?

**Comment by jtaala** [comment]
Anyone have any idea what `super+.` is supposed to do n terminals?  Here's in console using (comes up with an underlined `e` - kind of like switching languages or something):

https://github.com/paperwm/PaperWM/assets/30424662/61167ec2-4c81-4b01-b444-fa5084f6150d

**Comment by jtaala** [comment]
> But I'll try to disable keybinding and see what happens.

Yes please - try changing the keybind to confirm that it is that keybind in terminals (for some reason).  Will help debug and see what can be done about it.

**Comment by jtaala** [comment]
ah... it's to add emoji's and other characters... in the terminal.  Try `super`+`.` and then press spacebar.

https://github.com/paperwm/PaperWM/assets/30424662/e9f902f7-2ebe-4270-b828-07d49b38fd8e

**Comment by jtaala** [comment]
Okay, run `ibus-setup` from terminal and look at the keybind in the `Emoji` tab, change it to something else:

![image](https://github.com/paperwm/PaperWM/assets/30424662/9ae5324e-6f73-473c-99e9-10446e014f82)

**Comment by jtaala** [comment]
Dang, given it's ibus (input bus) - it can ignore paperwm keybinds in the terminal context (which is what it's doing here).  If that keybind was in dconf, then we could override it when PaperWM is active...

Not sure what we can do here - other than change the ibus emoji annotation keybind... or PaperWM's.

**Comment by jtaala** [comment]
Interestingly, people were complaining about this shortcut conflicting with a shortcut in vscode (doing the same as it is here in PaperWM).  That was when it was `cntrl+period`... looks like ibus changed it to `super+period` (?) to avoid that conflict... but now it conflicts with paperwm:

https://www.reddit.com/r/gnome/comments/v5p8fi/you_can_insert_emoji_quickly_by_pressing_ctrl/

**Comment by hpaul** [comment]
> ah... it's to add emoji's and other characters... in the terminal. Try super+. and then press spacebar.
Wow didn't expected that. How did you discovered it?

I changed these key combination and everything works as expected. :+1: . Thank you for the investigation. I was dealing with these for a few days.

I wonder why it started now, as I guess that UI for emojis was there before.

Again, thank you!!

**Comment by jtaala** [comment]
Actually I found the keybind preference in dconf (under `/desktop/ibus/panel/emoji/`) - so it might be worth adding this schema for conflict detection in paperwm (which could avoid this issue in the future for others).

**Comment by hpaul** [comment]
Although quite a nice useful thing for the future, mapped it to `Super+e` and works well.

**Comment by Lythenas** [comment]
> Anyone have any idea what `super+.` is supposed to do n terminals? Here's in console using (comes up with an underlined `e` - kind of like switching languages or something):
> Kooha-2023-11-15-10-59-56.mp4

Seems like you mostly figured it out, but that looks like the "compose key" (which can also be configured separately in the gnome settings). This allows you to input "composed" charactes. E.g. `Composekey`, `a`, `"` should produce `ä`. I use it on my UK layout keyboard to input german special characters.

I probably don't have ibus installed so I didn't run into this. At least I don't have the command `ibus-setup`.

I guess we can add `/desktop/ibus/panel/emoji/` to our list of controlled keybindings. But I'm a bit hesitant to add things outside of gnome to it.

**Comment by jtaala** [comment]
> Seems like you mostly figured it out, but that looks like the "compose key" (which can also be configured separately in the gnome settings

hmm - that compose key is disabled for me (not used) so I'm not sure that the key.  I tested adding the ibus dconf schema but it doesn't seem to detect the clash (not sure why there).

**Comment by fernandovmacedo** [comment]
Thanks for the tips, everyone! I was having this issue. Ran ibus-setup, changed the emoji shortcut there, and problem solved!

**Comment by ccope** [comment]
Just hit this after upgrading to Ubuntu 24.04. I think it would be a good idea to keep this issue pinned until there's a fix merged, it's very confusing to diagnose.
edit; the content of the Emoji keybinding in my ibus-setup window was `<Super>period; <Super>semicolon`. I had to hit the `...` button and then `Delete` on the `<Super>period` line.
